packages/neo-one-node-bin/src/startNode.ts

```diff
@@ -96,9 +89,12 @@ export const startNode = async (): Promise<void> => {
 
   const fullNode = new FullNode({
     monitor,
-    environment: createEnvironment(),
-    settings: createTest(),
-    options$: _of({}),
+    environment,
+    settings: {
+      ...createTest(),
```

Comment:
Not quite - don't merge some arbitrary `settings` object into the settings of `createTest`. Instead you should do something like
```
export interface NodeBinEnvironment {
  settings: SettingsEnvironment;
  fullNode: FullNodeEnvironment; 
}
```
then choose which of `createMain` or `createTest` to use based on a property in `SettingsEnvironment` and pass arguments from the properties of `SettingsEnvironment` to the function.